### PR TITLE
improve code-coverage testing of event scheduler

### PR DESCRIPTION
**Pull request by vtjnash**

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
You might want to mention that this is a fix for breakage caused by https://github.com/JuliaLang/julia/pull/21458. See also https://github.com/JuliaLang/julia/pull/21547 where I started to revert that PR (which I wouldn't have started on if there had been breadcrumbs to this PR).

**Comment by StefanKarpinski** [comment]
But overall, it's great to have better coverage of this, so all-in-all it's a good outcome.

**Comment by tkelman** [comment]
appveyor failures are real here

File: base/libc.jl [hunk 1]
```diff
@@ -26,16 +26,33 @@ Base.cconvert(::Type{Int32}, fd::RawFD) = fd.fd
...
+                (Int32, Int32), src.fd, target.fd))
 
 # Wrapper for an OS file descriptor (for Windows)
 if is_windows()
     struct WindowsRawSocket
         handle::Ptr{Void}   # On Windows file descriptors are HANDLE's and 64-bit on 64-bit Windows
     end
     Base.cconvert(::Type{Ptr{Void}}, fd::WindowsRawSocket) = fd.handle
-    _get_osfhandle(fd::RawFD) = WindowsRawSocket(ccall(:_get_osfhandle,Ptr{Void},(Cint,),fd.fd))
+    _get_osfhandle(fd::RawFD) = WindowsRawSocket(ccall(:_get_osfhandle, Ptr{Void}, (Cint, ), fd.fd))
```
Comment (tkelman) [inline base/libc.jl]: why the extra space here?
```
$ git grep ',)' | wc -l
1074

$ git grep ', )' | wc -l
30
```

Comment (vtjnash) [inline base/libc.jl]: Looks like all of the latter are simple formatting mistakes

Comment (tkelman) [inline base/libc.jl]: okay thanks, cleaned in 383e8d3b9f39933fd7aaf1b5f0bef6ba1f4724ed

**Comment by tkelman** [comment]
This didn't completely fix the breakage from https://github.com/JuliaLang/julia/pull/21458, there are still `UndefVarError`s in anything that was relying on the deprecation of consume
- https://github.com/JuliaCI/pkg.julialang.org/blob/8fd543df2ac929d0b4f55ca6531bb8e2c9eb3cac/logs/Erdos_0.6.log
- https://github.com/JuliaCI/pkg.julialang.org/blob/8fd543df2ac929d0b4f55ca6531bb8e2c9eb3cac/logs/Gumbo_0.6.log
- https://github.com/JuliaCI/pkg.julialang.org/blob/8fd543df2ac929d0b4f55ca6531bb8e2c9eb3cac/logs/NKLandscapes_0.6.log
- https://github.com/JuliaCI/pkg.julialang.org/blob/8fd543df2ac929d0b4f55ca6531bb8e2c9eb3cac/logs/PicoSAT_0.6.log
- https://github.com/JuliaCI/pkg.julialang.org/blob/8fd543df2ac929d0b4f55ca6531bb8e2c9eb3cac/logs/QDXML_0.6.log
- https://github.com/JuliaCI/pkg.julialang.org/blob/8fd543df2ac929d0b4f55ca6531bb8e2c9eb3cac/logs/StochasticSearch_0.6.log

**Pull request merged**